### PR TITLE
Only update test dependencies which reference specific repo built packages 

### PR DESCRIPTION
**Pull request by chcosta**
/cc @weshaggard @jhendrixMSFT

### CODE REVIEW

File: src/Microsoft.DotNet.Build.Tasks/PackageFiles/buildagainstpackages.targets [hunk 1]
```diff
@@ -182,6 +182,7 @@
                                   VersionStructureRegex="$(VersionStructureRegex)"
                                   AdditionalDependencies="@(_InjectProjectReferenceDependency)"
                                   ProjectJson="$(OriginalProjectJson)"
+                                  IdentityRegex="$(BuildTestsAgainstPackagesIdentityRegex)"
```
Comment (chcosta) [inline src/Microsoft.DotNet.Build.Tasks/PackageFiles/buildagainstpackages.targets]: I have an additional PR for the corefx repo which defines this property (regex).  I'll submit that PR after this PR is reviewed / merged.

**Comment by jhendrixMSFT** [comment]
Could you provide an example regex you'd pass to this task?

**Comment by chcosta** [comment]
I'll be passing in the existing corefx validation pattern - https://github.com/dotnet/corefx/blob/master/dir.props#L89

**Comment by weshaggard** [comment]
I'm currently OK with the approach but I'm starting to wonder how far these regex's will degenerate to such a point where we end up with a list of packages. Lets see how this goes.

**Comment by weshaggard** [comment]
LGTM

**Pull request merged**